core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/SparkRunner.scala

```diff
@@ -150,11 +150,22 @@ object SparkRunner extends Runner[CR] with PatchProvider[SpecPatch] {
       configMaps = configMaps,
       securityContext = securityContext
     )
-    val monitoring = Monitoring(prometheus = Prometheus(
-      jmxExporterJar = agentPaths(CloudflowApplication.PrometheusAgentKey),
-      configFile = PrometheusConfig.prometheusConfigPath(Runner.ConfigMapMountPath)
-    )
-    )
+
+    val monitoring = {
+      // if (agentPaths(CloudflowApplication.PrometheusAgentKey) == "/dummy") {
+      if (!agentPaths.contains(CloudflowApplication.PrometheusAgentKey)) {
+        Monitoring(prometheus = Prometheus(
+          jmxExporterJar = "/prometheus/jmx_prometheus_javaagent.jar",
+          configFile = "/etc/cloudflow-runner/prometheus.yaml",
+          port = 2050
```

Comment:
Is this a safe port? Btw here is the list of exporter ports out there: https://github.com/prometheus/prometheus/wiki/Default-port-allocations#exporters-outside-of-the-standard-port-ranges

Reply:
This I took from the earlier impl .. it has been 2050 right from beginning. But we can take a relook ..